AV-45749: Update sample queries (#35)
couchbaselabs/docs-devex: modules/n1ql/pages/n1ql-language-reference/selectintro.adoc

The optimizer attempts to select an appropriate secondary index for a query based on the filters in the WHERE clause.
If it cannot select a secondary query, the query service falls back on the primary index for the keyspace.

> By default, secondary indexes do not index a document if the leading index key is MISSING in the document.

Reviewer: ```suggestion
By default, secondary indexes don't index a document if the leading index key is MISSING in the document.
```

This means that when a query selects a field which is MISSING in some documents, the optimizer will not be able to choose a secondary index which uses that field as a leading key. There are two ways to resolve this:

> * You can add the `INCLUDE MISSING` modifier to a leading index key in the index definition, in order to index documents in which the specified key is missing. For more information, see xref:n1ql:n1ql-language-reference/createindex.adoc#include-missing[INCLUDE MISSING]

Reviewer: ```suggestion
* Add the `INCLUDE MISSING` modifier to a leading index key in the index definition, to index documents where the specified key is missing. For more information, see xref:n1ql:n1ql-language-reference/createindex.adoc#include-missing[INCLUDE MISSING]
```

> * You can use a WHERE clause of some kind to filter out any documents in which the required field is MISSING.

Reviewer: ```suggestion
* Use a WHERE clause to filter out any documents where the required field is MISSING.
```

The minimum filter you can use to do this is `IS NOT MISSING`.

> This is usually only necessary in queries which do not otherwise have a WHERE clause; for example, some GROUP BY and aggregate queries.

Reviewer 2: These are really important changes; INCLUDE MISSING needs to be properly incorporated into the examples. But it needs to be done properly, across all of release/7.2, capella, and elixir. Please revert this change and we'll address this in a separate PR which can be ported to other branches easily.

Author: Reverting

(((sargable)))
A query is said to be _sargable_ if the optimizer is able to select an index to speed up the execution of the query.

> For more examples, see xref:n1ql:n1ql-language-reference/groupby-aggregate-performance.adoc[Group By and Aggregate Performance].

Reviewer 2: Reinstate for now.

Author: Readdded

> For further examples, refer to xref:n1ql:n1ql-language-reference/groupby-aggregate-performance.adoc[Group By and Aggregate Performance].

Reviewer: ```suggestion
For more examples, see xref:n1ql:n1ql-language-reference/groupby-aggregate-performance.adoc[Group By and Aggregate Performance].
```
